src/UI/SignTransactionsModals/index.tsx

```diff
@@ -71,32 +72,71 @@ export const SignTransactionsModals = ({
     verifyReceiverScam
   };
 
-  if (signError || hasTransactions) {
+  const ConfirmScreens = {
+    Ledger: CustomConfirmScreens?.Ledger ?? SignWithLedgerModal,
+    WalletConnect:
+      CustomConfirmScreens?.WalletConnect ?? SignWithWalletConnectModal,
+    Extension: CustomConfirmScreens?.Extension ?? SignWithExtensionModal,
+    Extra: CustomConfirmScreens?.Extra ?? null
+  };
+
+  const renderTransactionStatusToast = () => {
+    if (signError) {
+      return (
+        <TransactionStatusToast
+          show={Boolean(signError)}
+          message={signError}
+          type={'error'}
+          onDelete={() => onAbort(sessionId)}
+        />
+      );
+    } else if (cancelTransactionsMessage) {
+      return (
+        <TransactionStatusToast
+          show={Boolean(cancelTransactionsMessage)}
+          message={cancelTransactionsMessage}
+          type={'warning'}
+          onDelete={() => onAbort(sessionId)}
+        />
+      );
+    }
+
+    return null;
+  };
+
+  const shouldShowTransactionStatusToast =
+    Boolean(signError) || Boolean(cancelTransactionsMessage);
+
+  if (shouldShowTransactionStatusToast || hasTransactions) {
     switch (loginMethod) {
       case LoginMethodsEnum.ledger:
-        return CustomConfirmScreens?.Ledger ? (
-          <CustomConfirmScreens.Ledger {...signProps} />
+        return shouldShowTransactionStatusToast ? (
+          renderTransactionStatusToast()
         ) : (
-          <SignWithLedgerModal {...signProps} />
+          <ConfirmScreens.Ledger {...signProps} />
         );
 
       case LoginMethodsEnum.walletconnect:
-        return CustomConfirmScreens?.WalletConnect ? (
-          <CustomConfirmScreens.WalletConnect {...signProps} />
+        return shouldShowTransactionStatusToast ? (
+          renderTransactionStatusToast()
         ) : (
-          <SignWithWalletConnectModal {...signProps} />
+          <ConfirmScreens.WalletConnect {...signProps} />
         );
 
       case LoginMethodsEnum.extension:
-        return CustomConfirmScreens?.Extension ? (
-          <CustomConfirmScreens.Extension {...signProps} />
+        return shouldShowTransactionStatusToast ? (
+          renderTransactionStatusToast()
         ) : (
-          <SignWithExtensionModal {...signProps} />
+          <ConfirmScreens.Extension {...signProps} />
         );
 
       case LoginMethodsEnum.extra:
-        return CustomConfirmScreens?.Extra ? (
-          <CustomConfirmScreens.Extra {...signProps} />
+        return ConfirmScreens?.Extra ? (
+          shouldShowTransactionStatusToast ? (
```

Comment:
I think I saw a comment left by you somewhere that double ternary are confusing :)

Reply:
right